### PR TITLE
Add isometria 75 keyboard

### PR DESCRIPTION
**Pull request by ebastler**
## Description

Adds support for the isometria 75 keyboard

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [X] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* Adds support for isometria 75 keyboard

## Checklist

- [X] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [X] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [X] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/ebastler/isometria_75/rev1/config.h [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0x6562 // "eb"
+#define PRODUCT_ID      0x1337
+#define DEVICE_VER      0x0001
+#define MANUFACTURER    ebastler
+#define PRODUCT         ebastler isometria 75
+#define DESCRIPTION     ebastler isometria 75
```
Comment (drashna) [inline keyboards/ebastler/isometria_75/rev1/config.h:28]:
```suggestion
#define PRODUCT        isometria 75
```

File: keyboards/ebastler/isometria_75/rev1/keymaps/default/keymap.c [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+#include "rev1.h"
```
Comment (drashna) [inline keyboards/ebastler/isometria_75/rev1/keymaps/default/keymap.c:17]:
```suggestion
```

File: keyboards/ebastler/isometria_75/rev1/keymaps/default/keymap.c [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,  KC_TRNS, KC_TRNS, MO(2),   KC_TRNS, KC_TRNS, 
+        KC_TRNS, KC_TRNS, KC_TRNS,                            KC_TRNS,                   KC_TRNS,  LT(1, KC_APP),    KC_TRNS, KC_TRNS, KC_TRNS),
+};
+
+/* Encoder */
+void encoder_update_user(uint8_t index, bool clockwise) {
+/* The first if reads the first encoder, not needed on this board which only features a single one */
+    if (index == 0) {
+        /* The switch case allows for different encoder mappings on different layers, "default" map gets applied for all unspecified layers */
+        switch(biton32(layer_state)){
```
Comment (drashna) [inline keyboards/ebastler/isometria_75/rev1/keymaps/default/keymap.c:51]:
```suggestion
        switch(get_highest_layer(layer_state)){
```

File: keyboards/ebastler/isometria_75/rev1/keymaps/via/keymap.c [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+#include "rev1.h"
```
Comment (drashna) [inline keyboards/ebastler/isometria_75/rev1/keymaps/via/keymap.c:17]:
```suggestion
```

File: keyboards/ebastler/isometria_75/rev1/keymaps/via/keymap.c [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,  KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, 
+        KC_TRNS, KC_TRNS, KC_TRNS,                            KC_TRNS,                   KC_TRNS,  KC_TRNS,          KC_TRNS, KC_TRNS, KC_TRNS),
+};
+
+/* Encoder */
+void encoder_update_user(uint8_t index, bool clockwise) {
+/* The first if reads the first encoder, not needed on this board which only features a single one */
+    if (index == 0) {
+        /* The switch case allows for different encoder mappings on different layers, "default" map gets applied for all unspecified layers */
+        switch(biton32(layer_state)){
```
Comment (drashna) [inline keyboards/ebastler/isometria_75/rev1/keymaps/via/keymap.c:59]:
```suggestion
        switch(get_highest_layer(layer_state)){
```

**Review by drashna — CHANGES_REQUESTED** [review]
Also, could you rename the readme files to be all lower case?  
This prevents issues with case sensitivity on certain OSes.

**Comment by ebastler** [comment]
Thank you for the check and corrections! I have accepted your changes, and renamed the readme filenames to lowercase.

File: keyboards/ebastler/isometria_75/rev1/rules.mk [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+# Build Options
+NKRO_ENABLE = yes		# USB Nkey Rollover - if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+EXTRAKEY_ENABLE = yes	# Audio control and System control(+450)
+RGBLIGHT_ENABLE = yes
+BACKLIGHT_ENABLE = yes
+ENCODER_ENABLE = yes
+WS2812_DRIVER = pwm
+BOOTMAGIC_ENABLE = lite # Enable lite mode of bootmagic (boot into BL)
+SLEEP_LED_ENABLE = no
+MOUSEKEY_ENABLE = no
```
Comment (fauxpark) [inline keyboards/ebastler/isometria_75/rev1/rules.mk:13]:
```suggestion
# Build Options
#   change yes to no to disable
#
BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration
MOUSEKEY_ENABLE = no        # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = yes           # USB Nkey Rollover
BACKLIGHT_ENABLE = yes      # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
ENCODER_ENABLE = yes
WS2812_DRIVER = pwm
```

File: keyboards/ebastler/isometria_75/rev1/rules.mk [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+
+# Enter lower-power sleep mode when on the ChibiOS idle thread
+OPT_DEFS += -DCORTEX_ENABLE_WFI_IDLE=TRUE
+OPT_DEFS += -DSTM32_DMA_REQUIRED=TRUE
```
Comment (fauxpark) [inline keyboards/ebastler/isometria_75/rev1/rules.mk:17]:
```suggestion
```
This is implied by `WS2812_DRIVER = pwm`.

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by ebastler** [comment]
Thank you very much for your corrections too (and Merry Christmas!). I have accepted them all, and will make future PRs correctly from the beginning. I just tested with all changes, it still builds and works as intended.

**Review by fauxpark — APPROVED** [review]

File: keyboards/ebastler/isometria_75/rev1/mcuconf.h [hunk 8]
```diff
@@ -0,0 +1,171 @@
+/*
```
Comment (drashna) [inline keyboards/ebastler/isometria_75/rev1/mcuconf.h:1]: Just a heads up, you don't need this entire board files here, Basically, you jus need this: 
https://github.com/qmk/qmk_firmware/blob/01c7108db52677c2a4210bcda0afd2b1d3f7619f/platforms/chibios/keyboard-config-templates/mcuconf.h

Just add the undef/define for any changes you need to make.

File: keyboards/ebastler/isometria_75/rev1/halconf.h [hunk 9]
```diff
@@ -0,0 +1,525 @@
+/*
```
Comment (drashna) [inline keyboards/ebastler/isometria_75/rev1/halconf.h:1]: Don't need all of this. 

Just need something like this: 

https://github.com/qmk/qmk_firmware/blob/01c7108db52677c2a4210bcda0afd2b1d3f7619f/platforms/chibios/keyboard-config-templates/halconf.h

File: keyboards/ebastler/isometria_75/rev1/chconf.h [hunk 10]
```diff
@@ -0,0 +1,714 @@
+/*
```
Comment (drashna) [inline keyboards/ebastler/isometria_75/rev1/chconf.h:1]: Don't need all of this: 

https://github.com/qmk/qmk_firmware/blob/01c7108db52677c2a4210bcda0afd2b1d3f7619f/platforms/chibios/keyboard-config-templates/chconf.h

File: keyboards/ebastler/isometria_75/rev1/mcuconf.h (hunk 8 above)
Comment (ebastler) [reply keyboards/ebastler/isometria_75/rev1/mcuconf.h:1]: Does that already work for F072 MCUs? I thought only F303s were supported by the minimal configs so far. I changed it and it throws a ton of errors during compiling now :/

**Comment by ebastler** [comment]
I changed the 3 chibi configs to minimal files, including only the necessary lines to get the board working, and a few changes I had to add in order to get it to compile (without really knowing why tbqh). Should be ready now.

**Review by drashna — APPROVED** [review]

**Pull request merged**